docs/book/src/cronjob-tutorial/pprof-tutorial.md

```diff
@@ -0,0 +1,68 @@
+# Monitoring Performance with Pprof
+
+`pprof` is a Go profiling tool integrated within the `controller-runtime` library, designed to help identify performance bottlenecks in areas such as CPU usage, memory allocation, and more. This profiling tool, from the Go package [`net/http/pprof`][pprof-go-docs], is especially useful for diagnosing performance issues by providing detailed insights into resource usage during the runtime of your application.
+
+`pprof` is integrated into the HTTP server used by the `controller-runtime` manager, allowing you to collect profiling data via HTTP endpoints. This data can then be visualized using `go tool pprof`. Pprof feature is built into the `controller-runtime` library, so there is no need to install it separately.
+
+The `controller-runtime` [Manager options][manager-options-doc] provide an easy way to enable `pprof`, allowing you to gather detailed runtime metrics for improving the performance of your controllers.
+
+For further information about the Pprof, check out the official [Github repository][github-repo].
+
+<aside class="warning">
+<h1>Pprof Not Recommended for Production</h1>
+
+While [Pprof][github-repo] is an excellent tool for profiling and debugging, it is not recommended to leave it enabled in production environments. The primary reasons are:
+
+1. **Security Risk**: The profiling endpoints expose detailed information about your application's performance and resource usage, which could be exploited if accessed by unauthorized users.
+2. **Overhead**: Continuous profiling can introduce performance overhead, especially under heavy load, potentially impacting production workloads.
+
+</aside>
+
+## How to use Pprof in your Codebase
+
+1. In your `cmd/main.go` file, add the following `controller-runtime` manager field to enable the `pprof` endpoints:
+
+    ```golang
+    mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+      ...
+      // PprofBindAddress is the TCP address that the controller should bind to
+      // for serving pprof. Specify the manager address and the port that should be bind.
+      PprofBindAddress:       ":8082",
+      ...
+    })
+    ```
+
+2. Start your controller on your localhost:
+
+    ```bash
+    make run
```

Comment:
we have more than one to run so , we should not add this info here
Who will looking for `Monitoring Performance with Pprof` is skilled in KB already. Anyway, we can link the Quick start, see: https://github.com/kubernetes-sigs/kubebuilder/pull/4160#discussion_r1758291378

Reply:
> Who will looking for Monitoring Performance with Pprof s skilled in KB already.

@camilamacedo86 perhaps the end user is skilled in K8s. However, I would like to keep things more explicit.

> Anyway, we can link the Quick start,

I think there's no need to have a separate quick start guide. The tutorial itself is so small, and straightforward that it make the separate quick start guide redundant. This is how I see.

WDYT?

Reply:
WDYT:

```
2. **Test It Out**

After enabling [Pprof][link-alias], you need to build and deploy your controller to test it out. Follow the steps in the [Quick Start guide][quick-start-run-it] to run your project locally or on a cluster.
[quick-start-run-it]: ./docs/book/src/quick-start.md#test-it-out.

Then, you can apply your CRs/samples in order to monitor the performance of its controllers.
```

Reply:
@camilamacedo86 looks OK.